Week 3: 김정은 과제1,2 제출

Jeong-Ag opened the pull request:
Feat: 3주차 수업 내용

Feat: SignUp 페이지 구현
-ConstraintLayout 구현
-Margin/padding 적용

Feat: 계산기 구현
-숫자 입력 후 사칙연산 기호 클릭 시에 하단에 결과 출력
-/ 의 경우 두 번째 숫자가 0이면 '-' 출력 후 안내메세지 출력

Jeong-Ag committed:
Week 3: 김정은 과제1,2 제출

Feat: 3주차 수업 내용
Feat: SignUp 페이지 구현
-ConstraintLayout 구현
-Margin/padding 적용
Feat: 계산기 구현
-숫자 입력 후 사칙연산 기호 클릭 시에 하단에 결과 출력
-/ 의 경우 두 번째 숫자가 0이면 '-' 출력 후 안내메세지 출력

Jeong-Ag committed:
Week 3: 김정은 과제 2 제출

Feat: 계산기 구현
-숫자 입력 후 사칙연산 기호 클릭 시에 하단에 결과 출력
-/ 의 경우 두 번째 숫자가 0이면 '-' 출력 후 안내메세지 출력

chattymin commented on 김정은/Week 3/hw1/app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,143 @@
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:layout_marginHorizontal="20dp"
+    android:layout_marginVertical="20dp"
+    android:paddingHorizontal="30dp"
+    android:paddingVertical="40dp"
```
마진과 패딩의 차이를 아시나요??
왜 분할해서 두 종류를 다 사용했는지 궁금합니다!

chattymin commented on 김정은/Week 3/hw1/app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,143 @@
+        android:layout_marginTop="30dp"
+        android:text="ID"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/textView2" />
```
작은 차이이긴 한데
"@+id/~"는 새로 생성한다는 의미이고
"@id/~"는 이미 있는걸 사용한다는 의미입니다.

위의 방법으로 해도 정상 작동하지만, 아래 방법으로 하는게 더 좋겠죠?

chattymin commented on 김정은/Week 3/hw2/app/src/main/java/com/jungag/hw2/MainActivity.kt:
```diff
@@ -0,0 +1,58 @@
+        binding.button1.setOnClickListener(View.OnClickListener {
+            val num1 = binding.num1.text.toString()
+            val num2 = binding.num2.text.toString()
+            if (num1.isNotEmpty() && num2.isNotEmpty()) {
```
오 공백검사까지
완벽 그자체 👍

하지만 빈칸(공백)이 들어간다면????
이것도 막을 수 있는 방법으로 바꾸면 더 좋을거 같아용

chattymin commented on 김정은/Week 3/hw2/app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,94 @@
+        app:layout_constraintHorizontal_bias="0.498"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/num2"
+        app:layout_constraintVertical_bias="0.049" />
```
이 코드의 경우 xml파일의 미리보기에만 반영되고, 실제 결과물에는 반영되지 않는 코드입니다.
불필요한 코드는 지우는게 좋을것 같아요~

chattymin reviewed (changes requested):
한주 고생하셨습니다!! 너무 잘하시네요 😆

Jeong-Ag committed:
week 3: 김정은 과제 1,2 리팩토링

Refactor: 제출 과제 피드백 반영 수정
-@+id -> @id로 수정
-isNotEmpty 함수 대신 inNotNullOrBlack 함수 사용하여 null,empty,공백 문자열 모두 검사
-필요없는 코드 제거
-변수명 수정

chattymin reviewed (approved):
열심히 수정하셨네요!! 아주그냥 완벽합니다 ㅎ
